src/AlternateLife.RageMP.Net/Scripting/ScriptingClasses/Commands.cs

```diff
@@ -1,157 +1,207 @@
+            return param.GetCustomAttributes(typeof (ParamArrayAttribute), false).Length > 0;
         }
-
-        private void RemoveCommands(Func<CommandInformation, bool> predicate)
+        
+        private readonly Dictionary<Type, ITypeParser> _typeParsingSwitch = new Dictionary<Type, ITypeParser>
         {
-            foreach (var command in _commands.Reverse())
+            {typeof(int),     new PrimitiveParser<int>(int.TryParse)},
+            {typeof(float),   new PrimitiveParser<float>(float.TryParse)},
+            {typeof(double),  new PrimitiveParser<double>(double.TryParse)},
+            {typeof(uint),    new PrimitiveParser<uint>(uint.TryParse)},
+            {typeof(long),    new PrimitiveParser<long>(long.TryParse)},
+            {typeof(char),    new PrimitiveParser<char>(char.TryParse)},
+            {typeof(bool),    new PrimitiveParser<bool>(bool.TryParse)},
+            {typeof(byte),    new PrimitiveParser<byte>(byte.TryParse)},
+            {typeof(sbyte),   new PrimitiveParser<sbyte>(sbyte.TryParse)},
+            {typeof(short),   new PrimitiveParser<short>(short.TryParse)},
+            {typeof(decimal), new PrimitiveParser<decimal>(decimal.TryParse)},
+            {typeof(ushort),  new PrimitiveParser<ushort>(ushort.TryParse)},
+            {typeof(ulong),   new PrimitiveParser<ulong>(ulong.TryParse)},
+            {typeof(string),  new StringParser()},
+            {typeof(Enum),    new EnumParser()}
+        };
+
+        private IEnumerable<object> ProcessArguments(IReadOnlyList<string> args, IReadOnlyList<ParameterInfo> expectedParameters)
+        {
+            if (args.Count < expectedParameters.Count)
+            {
+                return null;
+            }
+            var parsedArgs = new List<object>();
+            for (int i = 0; i < args.Count; i++)
             {
-                if (predicate(command.Value) == false)
+                var expectedType = expectedParameters[i].ParameterType;
+                var targetType = expectedType;
+                if (expectedType.BaseType == typeof(Enum))
                 {
-                    continue;
+                    expectedType = typeof(Enum);
                 }
+                bool isParams = IsParams(expectedParameters[i]);
 
-                _commands.TryRemove(command.Key, out _);
+                if (i == expectedParameters.Count - 1 && isParams)
+                {
+                    var paramList = new List<object>();
+                    for (; i < args.Count; i++)
+                    {
+                        paramList.Add(args[i]);
+                    }
+
+                    parsedArgs.Add(paramList.ToArray());
+                    break;
+                }
+
+                if (i == expectedParameters.Count - 1 && !isParams && args.Count > expectedParameters.Count)
+                {
+                    return null;
+                }
+                
+                if (!_typeParsingSwitch.ContainsKey(expectedType))
```

Comment:
These two check can be merged with this improved one:

```cs
if(_typeParsingSwitch.TryGetValue(expectedType, out var parser) == false || parser.TryParse(args[i], targetType, out var parsedParameter) == false) 
{
    return null;
}
```

Furthermore, try to write variable names completely. For example `parsedParameter` instead of `parsedPara`.